With a NATS server with JetStream disabled `jetstream` package functions always returns `context deadline exceeded`

### Observed behavior

When starting a NATS server with JetStream _disabled_, the `jetstream` package client operations always seem to return `context.deadlineExceededError` instead of specialized errors.

### Expected behavior

Functions like `AccountInfo(ctx context.Context)` should return `ErrJetStreamNotEnabled`.

### Server and client version

github.com/nats-io/nats.go v1.33.1
server version 2.10.11

### Host environment

go version go1.22.0 linux/amd64
Server running on Podman v4.8.3


### Steps to reproduce

Start Server with `podman run --rm -it -p 4222:4222 nats:latest`.

Quick and dirty client code:

```go
package main

import (
	"context"
	"testing"

	"github.com/nats-io/nats.go"
	"github.com/nats-io/nats.go/jetstream"
)

func TestNatsClient(t *testing.T) {
	nc, _ := nats.Connect("nats://localhost:4222")
	js, _ := jetstream.New(nc)
	_, err := js.AccountInfo(context.TODO())
	t.Error(err)
}
```

Output:

```text
--- FAIL: TestNatsClient (5.00s)
    .../client_test.go:15: context deadline exceeded
```

I also tried to unwrap and convert the error with no success.

--- a/jetstream/errors.go
+++ b/jetstream/errors.go
@@ -71,6 +71,10 @@ var (
 
 	// ErrJetStreamNotEnabled is an error returned when JetStream is not
 	// enabled.
+	//
+	// Note: This error will not be returned in clustered mode, even if each
+	// server in the cluster does not have JetStream enabled. In clustered mode,
+	// requests will time out instead.
 	ErrJetStreamNotEnabled JetStreamError = &jsError{apiErr: &APIError{ErrorCode: JSErrCodeJetStreamNotEnabled, Description: "jetstream not enabled", Code: 503}}
 
 	// ErrJetStreamNotEnabledForAccount is an error returned when JetStream is

--- a/jetstream/jetstream.go
+++ b/jetstream/jetstream.go
@@ -791,9 +791,11 @@ func validateSubject(subject string) error {
 
 // AccountInfo fetches account information from the server, containing details
 // about the account associated with this JetStream connection. If account is
-// not enabled for JetStream, ErrJetStreamNotEnabledForAccount is returned. If
-// the server does not have JetStream enabled, ErrJetStreamNotEnabled is
-// returned.
+// not enabled for JetStream, ErrJetStreamNotEnabledForAccount is returned.
+//
+// If the server does not have JetStream enabled, ErrJetStreamNotEnabled is
+// returned (for a single server setup). For clustered topologies, AccountInfo
+// will time out.
 func (js *jetStream) AccountInfo(ctx context.Context) (*AccountInfo, error) {
 	ctx, cancel := wrapContextWithoutDeadline(ctx)
 	if cancel != nil {

--- a/jserrors.go
+++ b/jserrors.go
@@ -22,6 +22,10 @@ var (
 	// API errors
 
 	// ErrJetStreamNotEnabled is an error returned when JetStream is not enabled for an account.
+	//
+	// Note: This error will not be returned in clustered mode, even if each
+	// server in the cluster does not have JetStream enabled. In clustered mode,
+	// requests will time out instead.
 	ErrJetStreamNotEnabled JetStreamError = &jsError{apiErr: &APIError{ErrorCode: JSErrCodeJetStreamNotEnabled, Description: "jetstream not enabled", Code: 503}}
 
 	// ErrJetStreamNotEnabledForAccount is an error returned when JetStream is not enabled for an account.

--- a/jsm.go
+++ b/jsm.go
@@ -288,9 +288,13 @@ type accountInfoResponse struct {
 	AccountInfo
 }
 
-// AccountInfo retrieves info about the JetStream usage from the current account.
-// If JetStream is not enabled, this will return ErrJetStreamNotEnabled
-// Other errors can happen but are generally considered retryable
+// AccountInfo fetches account information from the server, containing details
+// about the account associated with this JetStream connection. If account is
+// not enabled for JetStream, ErrJetStreamNotEnabledForAccount is returned.
+//
+// If the server does not have JetStream enabled, ErrJetStreamNotEnabled is
+// returned (for a single server setup). For clustered topologies, AccountInfo
+// will time out.
 func (js *js) AccountInfo(opts ...JSOpt) (*AccountInfo, error) {
 	o, cancel, err := getJSContextOpts(js.opts, opts...)
 	if err != nil {

